### PR TITLE
22276: Adds a Sphinx extension to automatically build feature attributes pages

### PR DESCRIPTION
**Pull request by cademack**

### CODE REVIEW

**Review by apbassett — APPROVED** [review]

File: source/_ext/build_engine_schema_docs.py [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+                "FeatureOriginalTypeString",
+                "FeatureOriginalTypeTime",
+                "FeatureOriginalTypeTimedelta",
+                "FeatureType",
+            ]
+
+            # THIS WILL NOT WORK IF HOWSO-ENGINE-PY IS INSTALLED EDITABLY
+            # (unless you have the .caml in your local repo too)
+            schema_map = get_api()['schemas']
+            os.makedirs("howso/types", exist_ok=True)
```
Comment (fulpm) [inline source/_ext/build_engine_schema_docs.py:43]: This needs to generate relative to the source directory of the app. Running `make html` locally generates this in the wrong location

File: source/_ext/build_engine_schema_docs.py [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+{% if 'any_of' in json_spec %}
+:One of:
+{% for option in json_spec.any_of %}
+- :doc:`{{ option.ref }}`
+{% endfor %}
+{% endif %}
+
+{% for k, v in json_spec.items() %}
+{% if k not in ['indices', 'any_of', 'type', 'description', 'additional_indices'] %}
+:{{ k }}: {{ v }}
```
Comment (fulpm) [inline source/_ext/build_engine_schema_docs.py:102]: When outputting values for type/enum which are a list, concatenate them using `|`

File: source/_ext/build_engine_schema_docs.py [hunk 3]
```diff
@@ -0,0 +1,193 @@
...
+        'parallel_write_safe': True,
+    }
+
+
+# Define our base template as a string
+BASE_TEMPLATE = """
+{{ title }}
+{{ '=' * title|length }}
+
+.. contents:: Table of Contents
```
Comment (fulpm) [inline source/_ext/build_engine_schema_docs.py:79]: Lets add `:backlinks: none` so the backwards links to the TOC are removed

File: source/_ext/build_engine_schema_docs.py [hunk 4]
```diff
@@ -0,0 +1,193 @@
...
+{{ json_spec.description }}
+{% endif %}
+
+{% if 'type' in json_spec %}
+:type: {{ 'object' if json_spec.type == 'assoc' else json_spec.type }}
+{% endif %}
+
+
+{% if 'any_of' in json_spec %}
+:One of:
```
Comment (fulpm) [inline source/_ext/build_engine_schema_docs.py:94]:
```suggestion
:One of:

```
Need a new line after this to fix a warning in the build output

File: source/_ext/build_engine_schema_docs.py (hunk 1 above)
Comment (fulpm) [reply source/_ext/build_engine_schema_docs.py:43]: Looks like you can use `app.srcdir`

**Review by fulpm — APPROVED** [review]

**Pull request merged**